### ISSUE TITLE
ppo后的效果还没有sft效果好

### ISSUE THREAD

**Issue by qinlodestar** [hiyouga/ChatGLM-Efficient-Tuning#113]
ppo：
<img width="849" alt="image" src="https://github.com/hiyouga/ChatGLM-Efficient-Tuning/assets/12521847/5a5b3761-b4ab-4261-a785-4af15d920710">

sft:
python cli_demo.py --checkpoint_dir=/root/autodl-tmp/ChatGLM-Efficient-Tuning/examples/path_to_sft_checkpoint
<img width="1035" alt="image" src="https://github.com/hiyouga/ChatGLM-Efficient-Tuning/assets/12521847/c55d8ae4-ac2d-45d7-910f-272cbbd88cd0">

是什么地方出错了呢？

**Comment by JoyDajunSpaceCraft** [comment]
这个是无监督学习啊 ppo，sft是有监督，ppo对期望值的输出是什么样的是不知道的，所以sft会比ppo效果好的

**Comment by hiyouga** [comment]
首先 PPO 训练应当基于 SFT 训练后的模型，不能直接对 ChatGLM-6B 原始模型做 PPO 训练。其次，在 PPO 训练时候应当观察 reward 曲线变化情况，如果 reward 没有上升那说明 PPO 训练没有效果。

**Comment by qinlodestar** [comment]
是用的SFT训练后的模型，reward的值上升不是很明显。
{'loss': 0.7099, 'reward': 7.6885, 'learning_rate': 8.707142611918962e-12, 'epoch': 1.0}
{'loss': 0.9642, 'reward': 8.8186, 'learning_rate': 3.737549350013581e-12, 'epoch': 1.0}
{'loss': 1.1333, 'reward': 10.8629, 'learning_rate': 8.386191874265592e-13, 'epoch': 1.0}

**Comment by hiyouga** [comment]
reward model 用哪个数据集训练的呢？

**Comment by qinlodestar** [comment]
CUDA_VISIBLE_DEVICES=0 python ../src/train_rm.py \
    --do_train \
    --dataset comparison_gpt4_zh \
    --dataset_dir ../data \
    --finetuning_type lora \
    --output_dir path_to_rm_checkpoint \
    --overwrite_cache \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --lr_scheduler_type cosine \
    --logging_steps 10 \
    --save_steps 1000 \
    --learning_rate 1e-5 \
    --num_train_epochs 1.0 \
    --plot_loss \
    --quantization_bit 8 \
    --fp16

**Comment by hiyouga** [comment]
现在的数据集质量不够好，因此会导致 PPO 效果差问题。
